Implement operator `as` in template language (#63)
rinja-rs/rinja: book/src/template_syntax.md

{% endif %}
```

### Type conversion

You can use the operator [`as`](https://doc.rust-lang.org/std/keyword.as.html) in `{{ … }}`
expressions, and `{% … %}` blocks. It works the same as in Rust, but with some deliberate
restrictions:

> - You can only use [primive types](https://doc.rust-lang.org/std/primitive/index.html)

Reviewer: ```suggestion
- You can only use [primitive types](https://doc.rust-lang.org/std/primitive/index.html)
```

  like `i32` or `f64` both as source variable type and as target type.

> - If the source is a reference to a primitive type, e.g. `&&&bool`, then rinja automatically

Reviewer: Still forgot backticks around `rinja`. :p

We do it in multiple places so for the power of coherency, we need to continue! 🤣

Author: But "rinja" is a name that we don't use as a code snippet in here. I would rather remove the backticks around the other instances in book/src/template_syntax.md:299 and README.md:3. Maybe replace it with `**Rinja**` in the book (start of a sentence; bold like the first use of a lemma in Wikipedia)?

Reviewer: We need to pick one, I'm fine with whichever.

> - If the source is a reference to a primitive type, e.g. `&&&bool`, then rinja automatically
>   dereferences the value until it gets get the unterlying `bool`.

Reviewer: ```suggestion
- If the source is a reference to a primitive type, e.g. `&&&bool`, then rinja automatically
  dereferences the value until it gets the underlying `bool`.
```

> - If the source is a reference to a primitive type, e.g. `&&&bool`, then rinja automatically
>   dereferences the value until it gets get the underlying `bool`.

Reviewer: ```suggestion
- If the source is a reference to a primitive type, e.g. `&&&bool`, then `rinja` automatically
  dereferences the value until it gets the underlying `bool`.
```

Some parts of the suggestion were skipped. ;)

Author: Oops, fixed now. Thank you for reading my texts, twice. :D
